### DISCUSSION TITLE
How to select GPU?

### DISCUSSION THREAD

**Discussion by kpagcha** [lllyasviel/Fooocus#1151] [Q&A] [answered]
I am running this on Windows but when I check the GPU usage, it seems to generally prefer to use the Intel GPU instead of my Geforce one. I changed the bat to this setting I found:

`set TF_CUDNN_USE_AUTOTUNE=0 CUDA_VISIBLE_DEVICES=2 & .\python_embeded\python.exe -s Fooocus\entry_with_update.py`

I tried `CUDA_VISIBLE_DEVICES=1` too.

But it keeps using mostly the Intel GPU.

**Comment by robinpoppins** [comment]
I have the same issue. The programm runs on my internal gpu, but i want to use my rtx 4060. i got the latest drivers. can anyone please assist with code adaptions?

**Reply by dawidteo** [reply, marked as answer]
I added in "Launch options" in "Extra Launch Arguments" "--gpu-device-id  1" and it works great. Use two GTX1070 in two Fooocuses apps in the same time.

**Reply by SreeisonGithub** [reply]
can anyone tell me where to add this, like i am a beginner and pls explain in laymen terms as to where i can add launch options, i run foocus and saw that gpu usage was around 4 percent, and then it crashes.

**Reply by Rogala** [reply]
Edit the run.bat launch file in Notepad or another editor
[All CMD Flags](https://github.com/lllyasviel/Fooocus#all-cmd-flags)

**Reply by Ni-Deshmane** [reply]
Hi, can someone paste the completed code for this? I'm lost on how to modify the files without breaking something

**Reply by fparie** [reply]
this option does not seem to do anything when using AMD GPUs and the --directml option. Is it possible to select AMD GPU ID 1 ?

**Reply by NoamTheStudent** [reply]
I also tried to do it with my amd pc. the program still insist on using the integratef GPU and not the dedicated one (AMD RX 7700 XT). The program doesn't work at all like this.

**Comment by robinpoppins** [comment]
Awesome, thank you!

dawidteo ***@***.***> schrieb am Do., 1. Feb. 2024, 01:12:

**Comment by jimmyandrews** [comment]
Not sure if this is appropriate to reopen this issue, but I'm having some oddities with the Docker version of the deployment and this issue. 

I found where to add in that setting (docker-compose.yml, Line 11, - CMDARGS=...) and can successfully rerun docker compose to push up that config. During the container startup, everything looks good, sees the new flag, finds my NVidia GPU, sets it up, recognizes it correctly, etc. (Turns out with the default settings it was already grabbing the NVidia GPU anyway.) But when I go to run a prompt, I can see the NVidia VRAM being exercised, a blip on the 3D use when the prompt is first executed, but then it's the Intel GPU that does all the 3D processing. It was most easily to see the behavior using the "default" preset and using only the SAI 3D Model style, with a prompt of "Beautiful pasture at sunset."

What am I missing, or is this standard behavior?

**Reply by pawrequest** [reply]
same here, but launching from the provided .bat files. 

console output correctly identifies the nvidia GPU but task-manager shows work is done on integrated AMD gpu.

**Reply by jimmyandrews** [reply]
I've poked around a bit with this and apparently there is some bugginess with the Task Manager and showing proper values. When I open another application that pings the GPU usage, it does appear that my NVidia GPU is being exercised.

I'm starting to play with stable-diffusion-webui-forge code, now that I have my bearings a bit more. Updated my CUDA drivers too using the NVidia CUDA installer, so hopefully I can have some luck there.
